File: problems/1059/main.c
#include <stdio.h>
#include <gmp.h>

#define MIN 0x10000
#define MAX 0xFFFFF

int main() {
    unsigned int x, y, z, count;
    char buffer[8];
    mpz_t result;

    mpz_init(result);

	x = MIN;
	for (y = x+1; y < MAX; y++) {
		z = x+y;
		if (z > MAX) {
			break;
		}
		sprintf(buffer, "%x", z);
		if (buffer[4] == '1') {
			count++;
		}
	}

	mpz_add_ui(result, result, count);

	while (count > 0) {
		x++;
		sprintf(buffer, "%x", x);
		if (buffer[4] == '1' || buffer[4] == '9') {
			count--;
		}
		mpz_add_ui(result, result, count);
	}

	gmp_printf("answer is %Zd\n", result);

	mpz_clear(result);
	return 0;
}

File: problems/1059/main.js
const MIN = 0x10000;
const MAX = 0xfffff;

const fifthDigit = number => number.toString(16)[4];

let accepted = 0;

let a = MIN;

for (let b = a + 1; b <= MAX; b++) {
	const c = a + b;

	if (c > MAX) {
		break;
	}

	if (fifthDigit(c) == '1') {
		accepted += 1;
	}
}

let possible = accepted;

while (possible > 0) {
	a += 1;

	if (['1', '9'].includes(fifthDigit(a))) {
		possible -= 1;
	}

	accepted += possible;
}

console.log(accepted);
